selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/Editor.java

```diff
@@ -0,0 +1,1914 @@
+        waitActiveEditor();
+        WebElement currentActiveElement =
+                activeLineNumbers.stream().filter(webElement -> webElement.isDisplayed()).findFirst().get();
+
+        return getCursorPositionFromWebElement(currentActiveElement);
+    }
+
+    /**
+     * Check that editor active and focused on first step. After that get cursor position. Note! For correct work a editor must be active.
+     *
+     * @return char and line position from current visible, active and focused editor (usual uses with split editor)
+     */
+    public Pair<Integer, Integer> getCursorPositionsFromCurentActiveEditor() {
+        String xpathToCurrentActiveCursorPosition =
+                "//div[@active and @focused]/parent::div[@id='gwt-debug-multiSplitPanel-tabsPanel']/div[@active and @focused]/parent::div[@id='gwt-debug-multiSplitPanel-tabsPanel']/parent::div/parent::div/following-sibling::div//div[@active]//div[@id='gwt-debug-cursorPosition']";
+        waitActiveEditor();
+        return getCursorPositionFromWebElement(
+                redrawDriverWait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(xpathToCurrentActiveCursorPosition))));
+
+    }
+
+    private Pair<Integer, Integer> getCursorPositionFromWebElement(WebElement webElement) {
+        int[] currentCursorPositions =
+                Arrays.asList(webElement.getText().split(":")).stream().mapToInt(Integer::parseInt).toArray();
+        return new Pair<Integer, Integer>(currentCursorPositions[0], currentCursorPositions[1]);
+    }
+
+    /** get number of current line from visible editor. If editor split - can be some visible items, in this case will be returned firs in the DOM @return line value */
+    public int getPositionOfLineFromVisibleEditor() {
+        waitActiveEditor();
+        return getCurrentCursorPositionsFromVisibleEditor().first;
+    }
+
+    /**
+     * check that visible editor line has expected number of line
+     *
+     * @param expectedLine
+     *         expected number of visible line
+     */
+    public void waitExpectedNumberOfActiveLine(final int expectedLine) {
+        waitActiveEditor();
+        redrawDriverWait.until(
+                (ExpectedCondition<Boolean>)driver -> expectedLine == getPositionOfLineFromVisibleEditor());
+    }
+
+    /**
+     * get number of current char from visible editor. If editor split - can be some visible items, in this case will be returned firs in the DOM
+     *
+     * @return char value
+     */
+    public int getPositionOfCharFromVisibleEditor() {
+        return getCurrentCursorPositionsFromVisibleEditor().second;
+    }
+
+    /**
+     * wait specified values for Line and Char cursor positions in the Codenvy editor
+     *
+     * @param linePosition
+     *         expected line position
+     * @param charPosition
+     *         expected char position
+     */
+    public void waitLineAndCharForCurrentVisibleEditor(final int linePosition, final int charPosition) {
```

Comment:
IMHO, it's better to rename the method to have a shorter name **waitCursorInPosition()** .